src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -1209,9 +1237,14 @@ private void checkpointIfBatchMode() {
 			}
 		}
 
-		private void checkpointIfRecordMode(Record record) {
+		private void checkpointIfRecordMode(@Nullable Record record) {
 			if (CheckpointMode.record.equals(KinesisMessageDrivenChannelAdapter.this.checkpointMode)) {
-				this.checkpointer.checkpoint(record.getSequenceNumber());
+				if (record == null) {
+					this.checkpointer.checkpoint();
+				}
+				else {
+					this.checkpointer.checkpoint(record.getSequenceNumber());
+				}
```

Comment:
Since we are going to call `checkpointer.checkpoint(endingSequenceNumber)` directly this change becomes redundant